### ISSUE TITLE
Markdown Characters Should Be Skipped When Reading Aloud

### ISSUE THREAD

**Issue by Mark-Phillipson** [microsoft/vscode#235295]
Type: <b>Bug</b>

**Title:** Markdown Characters Should Be Skipped When Reading Aloud

**Description:**
When reading back Markdown content, the reader currently vocalizes Markdown syntax characters such as asterisks (`*`) used for bold or italic text. This behavior detracts from the meaning and readability of the content. The reader should skip over Markdown characters and only vocalize the actual text content.

**Steps to Reproduce:**
1. Create a Markdown file with the following content:
    ```markdown
    **Bold Text**
    *Italic Text*
    ```
2. Use the text-to-speech feature to read the content aloud.

**Expected Behavior:**
The reader should vocalize:
- "Bold Text"
- "Italic Text"

**Actual Behavior:**
The reader vocalizes:
- "Asterisk asterisk Bold Text asterisk asterisk"
- "Asterisk Italic Text asterisk"

**Environment:**
- Application: Github Copilot Chat
- Version: [Version Number]
- Operating System: Windows 11

**Additional Information:**
Skipping Markdown characters will improve the clarity and comprehension of the read-aloud content.

VS Code version: Code 1.95.3 (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i9-12900K (24 x 3187)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.73GB (9.13GB free)|
|Process Argv|C:\\Users\\MPhil\\AppData\\Roaming\\talon\\user\\mystuff\\talon_my_stuff\\apps\\talon_my_stuff.code-workspace --crash-reporter-id 20d139f4-04aa-49d2-afa0-4b3653a32416|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
numbered-bookmarks|ale|8.5.0
andreas-talon|And|3.54.0
github-markdown-preview|bie|0.3.0
markdown-checkbox|bie|0.4.0
markdown-emoji|bie|0.3.0
markdown-footnotes|bie|0.1.1
markdown-mermaid|bie|1.27.0
markdown-preview-github-styles|bie|2.1.0
markdown-yaml-preamble|bie|0.1.0
vscode-eslint|dba|3.0.10
gitlens|eam|16.0.4
vscode-firefox-debug|fir|2.9.11
codespaces|Git|1.17.3
copilot|Git|1.246.1240
copilot-chat|Git|0.22.4
vscode-github-actions|git|0.27.0
vscode-pull-request-github|Git|0.100.3
vscode-peacock|joh|4.2.2
rainbow-csv|mec|3.13.0
vscode-talonscript|mro|0.3.22
debugpy|ms-|2024.12.0
python|ms-|2024.20.0
vscode-pylance|ms-|2024.11.3
remote-ssh-edit|ms-|0.87.0
extension-test-runner|ms-|0.0.12
vscode-speech|ms-|0.12.1
vscode-speech-language-pack-en-gb|ms-|0.5.0
talon-filetree|Pau|0.6.9
material-icon-theme|PKi|5.14.1
command-server|pok|0.10.1
cursorless|pok|0.29.1301
parse-tree|pok|0.32.0
semantic-movement|pok|0.3.0
talon|pok|0.2.0
vscode-xml|red|0.27.2
todotasks|san|0.5.0
vscode-fileutils|sle|3.10.3
errorlens|use|3.20.0
talonfmt-vscode|wen|0.1.1
markdown-all-in-one|yzh|3.6.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc2:31192216
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530

```

</details>

**Comment by meganrogge** [comment]
I'm not sure that is what we want because screen reader users might want to hear about the asterisks.

**Comment by Mark-Phillipson** [comment]
Coming from a nones screen reader user I am a non keyboard or mouse user when hearing back an article and every time it hits a word that is to be bolded it says asterix asterix then the word then asterix asterix when you're trying to understand what's being said this is very off putting.

I'm not sure how this is handled in a website for screen readers, but I would suggest that the asterix asterix be removed and the word be said in a different tone or pitch to indicate that it is bolded. This would make it easier to understand what is being said and not be so off putting when hearing it.

**Comment by Mark-Phillipson** [comment]
I still believe this is a bug by the way 👍🏻
